### PR TITLE
Removed redundant member variables

### PR DESCRIPTION
**Pull request by unnonouno**
I removed redundant member variables and instead fixed the function to use retained inputs/outputs.

### CODE REVIEW

File: chainer/functions/connection/n_step_rnn.py [hunk 1]
```diff
@@ -415,14 +414,15 @@ def forward(self, inputs):
                 self.reserve_space.data.ptr, reserve_size)
 
         self.c_y_descs = c_y_descs
-        self.ys = ys
         self.c_x_descs = c_x_descs
 
         if self.use_cell:
             # LSTM
+            self.retain_outputs([2])
```
Comment (okuta) [inline chainer/functions/connection/n_step_rnn.py]: How about using tuple to improve performance? `(2,)`

File: chainer/functions/connection/n_step_rnn.py [hunk 2]
```diff
@@ -415,14 +414,15 @@ def forward(self, inputs):
...
             return hy, cy, ys
         else:
             # GRU, RNN
+            self.retain_outputs([1])
```
Comment (okuta) [inline chainer/functions/connection/n_step_rnn.py]: ditto.

**Comment by unnonouno** [comment]
fixed

**Comment by okuta** [comment]
LGTM!

**Pull request merged**